src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/shared/packet/Vxlan.java

```diff
@@ -0,0 +1,75 @@
+/* Copyright 2017 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.floodlight.shared.packet;
+
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import net.floodlightcontroller.packet.BasePacket;
+import net.floodlightcontroller.packet.IPacket;
+
+import java.nio.ByteBuffer;
+
+@lombok.Data
+@EqualsAndHashCode(callSuper = true)
+@NoArgsConstructor
+public class Vxlan extends BasePacket {
+    private static final byte FLAGS = 0x8;
+    private static final int VXLAN_HEADER_SIZE_IN_BYTES = 8;
+    private static final int FIRST_RESERVED_AREA_IN_BYTES = 3;
+    private static final int SECOND_RESERVED_AREA_IN_BITS = 8;
+
+    private int vni;
+
+    @Override
+    public byte[] serialize() {
```

Comment:
Totally unreadable. You fill only one field and name all other as "reserved"... ugly. Without `VxLan` packet schema somewhere in front of eyes this code is not modifiable.

Reply:
Added description.